### PR TITLE
Support parse data

### PR DESCRIPTION
**Pull request by ogibson**
# Description

# Details
#### URL to issue
https://github.com/hms-dbmi-cellenics/issues/issues/104

#### Link to staging deployment URL (or set N/A)
N/A
https://ui-martinfosco-ui66-api28-pip.scp-staging.biomage.net/

#### Links to any PRs or resources related to this PR
https://github.com/hms-dbmi-cellenics/pipeline/pull/350
https://github.com/hms-dbmi-cellenics/api/pull/518
https://github.com/hms-dbmi-cellenics/ui/pull/965

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [x] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [x] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [x] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [x] Started end-to-end tests on the latest commit.

### Documentation updates
- [x] Relevant Github READMEs updated **or** no GitHub README updates required.
- [x] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: pipeline-runner/R/gem2s-2-load_user_files.R [hunk 1]
```diff
@@ -45,6 +46,64 @@ load_user_files <- function(input, pipeline_config, prev_out, input_dir = INPUT_
...
+
+    annotations <- read_parse_annotations(annot_fpath, sample)
+
+    mtx <- file.path(sample_dir, "DGE.mtx.gz")
+    cells <- file.path(sample_dir, "cell_metadata.csv.gz")
+    features <- file.path(sample_dir, "all_genes.csv.gz")
+
+    counts <- Seurat::ReadMtx(mtx = mtx, cells = cells, features = features,
+                   cell.column = 1, feature.column = 1, cell.sep = ",",
+                   feature.sep = ",", skip.cell = 1, skip.feature = 1, mtx.transpose = TRUE)
```
Comment (gerbeldo) [inline pipeline-runner/R/gem2s-2-load_user_files.R:86]: why are we not using Seurat::ReadParseBio (sure, it calls readMtx), I think there was a reason for it?

File: pipeline-runner/R/gem2s-2-load_user_files.R [hunk 2]
```diff
@@ -20,12 +20,13 @@ load_user_files <- function(input, pipeline_config, prev_out, input_dir = INPUT_
   # destructure previous output
   config <- prev_out$config
 
-  technology <- ifelse(config$input$type %in% c("rhapsody","10x_h5"),config$input$type , "10x")
+  technology <- ifelse(config$input$type %in% c("rhapsody","10x_h5","parse"),config$input$type , "10x")
```
Comment (gerbeldo) [inline pipeline-runner/R/gem2s-2-load_user_files.R:23]:
```suggestion
  technology <- ifelse(config$input$type %in% c("rhapsody", "10x_h5", "parse"), config$input$type , "10x")
```

File: pipeline-runner/R/gem2s-2-load_user_files.R [hunk 3]
```diff
@@ -45,6 +46,64 @@ load_user_files <- function(input, pipeline_config, prev_out, input_dir = INPUT_
...
 
+#' Read Parse files
+#'
+#' @param config
+#' @param input_dir
+#'
+#' @return
+#' @export
+#'
+#' @examples
```
Comment (gerbeldo) [inline pipeline-runner/R/gem2s-2-load_user_files.R:57]: document, remove @examples line

File: pipeline-runner/R/gem2s-2-load_user_files.R [hunk 4]
```diff
@@ -275,6 +334,21 @@ parse_rhapsody_matrix <- function(config, input_dir) {
...
 
+read_parse_annotations <- function(annot_fpath, sample){
+  annot <- read.csv(annot_fpath, header = TRUE)
+
+  # Make the names in annot the same as the ones in the Read10x generated count matrix
+  # Since Seurat uses makes.unique, we need to as well.
+  # Only for the first column, at this stage column 1 are the counts matrix rownames.
+  annot[,1] <- make.unique(annot[,1])
+
+  # Equalizing number of columns in case there's no Gene Expression column
```
[the remark this answers is not in the document]
Comment (gerbeldo) [reply pipeline-runner/R/gem2s-2-load_user_files.R:345]: the [all_genes files contain the gene name, ID and genome](https://arc.net/l/quote/eczwnnxm). we will not use this column for now, so it makes sense to remove it. update comment then.

File: pipeline-runner/R/gem2s-2-load_user_files.R [hunk 5]
```diff
@@ -45,6 +46,64 @@ load_user_files <- function(input, pipeline_config, prev_out, input_dir = INPUT_
...
+    message(
+      "Reading files from ",
+      sample_dir,
+      " --> ",
+      paste(sample_fpaths, collapse = " - ")
+    )
+
+    annotations <- read_parse_annotations(annot_fpath, sample)
+
+    mtx <- file.path(sample_dir, "DGE.mtx.gz")
```
Comment (gerbeldo) [inline pipeline-runner/R/gem2s-2-load_user_files.R:80]: rename to something more consistent? matrix_path or counts_path

File: pipeline-runner/R/gem2s-2-load_user_files.R [hunk 6]
```diff
@@ -45,6 +46,64 @@ load_user_files <- function(input, pipeline_config, prev_out, input_dir = INPUT_
...
+    cells <- file.path(sample_dir, "cell_metadata.csv.gz")
+    features <- file.path(sample_dir, "all_genes.csv.gz")
```
Comment (gerbeldo) [inline pipeline-runner/R/gem2s-2-load_user_files.R:82]: these as well. cells -> barcodes for example, to be consistent with naming in the other read_* functions

File: pipeline-runner/R/gem2s-2-load_user_files.R [hunk 7]
```diff
@@ -45,6 +46,64 @@ load_user_files <- function(input, pipeline_config, prev_out, input_dir = INPUT_
...
+
+    counts <- Seurat::ReadMtx(mtx = mtx, cells = cells, features = features,
+                   cell.column = 1, feature.column = 1, cell.sep = ",",
+                   feature.sep = ",", skip.cell = 1, skip.feature = 1, mtx.transpose = TRUE)
+
+    message(
+      sprintf(
+        "Sample %s has %s genes and %s droplets.",
```
Comment (gerbeldo) [inline pipeline-runner/R/gem2s-2-load_user_files.R:90]: not droplets! change to "barcodes"?

File: pipeline-runner/R/gem2s-2-load_user_files.R [hunk 8]
```diff
@@ -45,6 +46,64 @@ load_user_files <- function(input, pipeline_config, prev_out, input_dir = INPUT_
...
+#' @return
+#' @export
+#'
+#' @examples
+read_parse_files <- function(config, input_dir) {
+  counts_list <- list()
+  annot_list <- list()
+  feature_types_list <- list()
+
+  samples <- config$samples
```
Comment (gerbeldo) [inline pipeline-runner/R/gem2s-2-load_user_files.R:63]: Does parse data come de-multiplexed? one sample per folder? or are we offloading that work to users? What does the seurat function do about this?

Comment (gerbeldo) [reply pipeline-runner/R/gem2s-2-load_user_files.R:63]: eventually we might need to use the sample column in the `cell_metadata.csv` file. but we will need to think more about this.

File: pipeline-runner/R/gem2s-2-load_user_files.R [hunk 9]
```diff
@@ -45,6 +46,64 @@ load_user_files <- function(input, pipeline_config, prev_out, input_dir = INPUT_
...
+    c(counts, annotations) %<-% filter_unnamed_features(counts, annotations, sample)
+
+    counts_list[[sample]] <- counts
+    annot_list[[sample]] <- annotations
+  }
+
+  annot <- format_annot(annot_list)
+
+  return(list(counts_list = counts_list, annot = annot))
+
```
Comment (gerbeldo) [inline pipeline-runner/R/gem2s-2-load_user_files.R:104]:
```suggestion
```

File: pipeline-runner/R/gem2s-2-load_user_files.R (hunk 2 above)
Comment (gerbeldo) [reply pipeline-runner/R/gem2s-2-load_user_files.R:23]: spacing is wrong

File: pipeline-runner/R/gem2s-2-load_user_files.R (hunk 8 above)
Comment (ogibson) [reply pipeline-runner/R/gem2s-2-load_user_files.R:63]: Every parse experiment I've seen so far comes separated in folders. At some point in the future we could implement support for the sample column in cell_metadata but I think we need to learn more about the format beforehand.

File: pipeline-runner/R/gem2s-2-load_user_files.R (hunk 1 above)
Comment (ogibson) [reply pipeline-runner/R/gem2s-2-load_user_files.R:86]: Added a comment

File: pipeline-runner/R/gem2s-2-load_user_files.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (ogibson) [reply pipeline-runner/R/gem2s-2-load_user_files.R:345]: For now we use only the first two columns and not the genome, so yes.

**Review by gerbeldo — APPROVED** [review]
lgtm

File: pipeline-runner/tests/testthat/test-gem2s-2-load_user_files_parse.R [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+  )
+
+  # ensembl ids are in column 'input' of annot
+  expect_equal(out$annot$input, features$gene_id)
+
+  # symbols are in column 'name' of annot
+  expect_equal(out$annot$name, features$gene_name)
+})
+
+test_that("load_user_files loads 10x multisample experiments", {
```
Comment (saracastel) [inline pipeline-runner/tests/testthat/test-gem2s-2-load_user_files_parse.R:98]: shouldn't it be: loads `parse` multisample experiments?

**Pull request merged**